crux-core/src/crux/api/v2/EvictOperation.java

```diff
@@ -0,0 +1,58 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentVector;
+
+import java.util.Date;
+
+// TODO: Only allow withEndValidTime, keepLatest and keepEarliest if previous is set
+
+public class EvictOperation extends Operation {
+    private final Date validTime;
+    private final Date endValidTime;
+    private final Boolean keepLatest;
+    private final Boolean keepEarliest;
+    private final CruxId evictId;
+
+    private EvictOperation(CruxId evictId, Date validTime, Date endValidTime, Boolean keepLatest, Boolean keepEarliest) {
+        this.evictId = evictId;
+        this.validTime = validTime;
+        this.endValidTime = endValidTime;
+        this.keepLatest = keepLatest;
+        this.keepEarliest = keepEarliest;
+    }
+
+    public static EvictOperation evictOp(CruxId evictId) {
+        return new EvictOperation(evictId, null, null, null, null);
+    }
+
+    public EvictOperation withValidTime(Date validTime) {
+        return new EvictOperation(evictId, validTime, endValidTime, keepLatest, keepEarliest);
+    }
+
+    public EvictOperation withEndValidTime(Date endValidTime) {
+        return new EvictOperation(evictId, validTime, endValidTime, keepLatest, keepEarliest);
+    }
+
+    public EvictOperation keepLatest(Boolean keepLatest) {
+        return new EvictOperation(evictId, validTime, endValidTime, keepLatest, keepEarliest);
+    }
+
+    public EvictOperation keepEarliest(Boolean keepEarliest) {
+        return new EvictOperation(evictId, validTime, endValidTime, keepLatest, keepEarliest);
+    }
+
+    @Override
+    protected PersistentVector toEdn() {
+        PersistentVector outputVector = PersistentVector.create(Keyword.intern("crux.tx/evict"), evictId.toEdn());
```

Comment:
Oh, this looks very dangerous :fire:.
What happens if `validTime` is not specified, but `endValidTime` is? `endValidTime` will be used instead!
No-no, generic builder pattern is not really applicable to this class. 
Both timestamps can only come together in a single method call.

It could be something like 
```
EvictOperation.create(myId).validTimes(from, to).keepLatest()
```

Reply:
Mhm, this was another one of the issues we were going through this morning. Evict has some planned changes, so at this stage we were going to remove the time ranges / keepearliest and keeplatest from this part of the API and re-add them down the line :)

Reply:
yep, to confirm, have removed the time-based flags to evict in line with the upcoming upstream changes